src/sources/datadog_agent/logs.rs

```diff
@@ -215,28 +214,24 @@ pub(crate) fn decode_log_body(
 // the tag list members are separated by `,` and the
 // tag-value pairs are separated by `:`.
 //
-// The output is an Object regardless of the input string.
-// Bare tags are constructed as a k-v pair with a null value.
+// The output is an Array regardless of the input string.
 fn parse_ddtags(ddtags_raw: &Bytes) -> Value {
     if ddtags_raw.is_empty() {
-        return ObjectMap::new().into();
+        return Vec::<Value>::new().into();
     }
 
     let ddtags_str = String::from_utf8_lossy(ddtags_raw);
 
     // The value is a single bare tag
     if !ddtags_str.contains(',') && !ddtags_str.contains(':') {
-        return ObjectMap::from([(KeyString::from(ddtags_str), Value::Null)]).into();
+        return vec![Value::Bytes(ddtags_raw.clone())].into();
     }
```

Comment:
Nit: is this short-circuiting significant processing? It seems like letting it fall through to the split below would have the same result with minimal overhead.